src/ui-kit/experimental/hierarchical/hierarchical-tree-selectedItem.model.ts

```diff
@@ -1,38 +1,86 @@
 export class HierarchicalTreeSelectedItemModel {
-    public treeMode: TreeMode;
+    /**
+     * 
+     */
+    public treeMode: TreeMode = TreeMode.SINGLE;
+
+    /**
+     * 
+     */
     private items: object[];
 
+    /**
+     * 
+     */
     constructor() {
         this.items = [];
     }
 
+    /**
+     * adds an item to the collection
+     * if tree mode is single it removes any existing items
+     * also checks to see if that item already exists
+     * keyfield is used to determine uniqueness of the item added
+     * @param item 
+     * @param keyField 
+     */
     addItem(item: object, keyField: string) {
         if (!this.contatinsItem(item[keyField], keyField)) {
             if (this.treeMode === TreeMode.SINGLE) {
                 this.clearItems();
             }
             this.items.push(item);
-            console.log('Item Added');
-            console.log(item);
         }
     }
 
+    /**
+     * adds an array of items to the list and will not add duplicate items
+     * keyfield is used to determine uniqueness of the item added
+     * @param items 
+     * @param keyField 
+     */
+    addItems(items: object[], keyField: string) {
+        for (let i = 0; i < items.length; i++) {
+            this.addItem(items[i], keyField);
+        }
+    }
+
+    /**
+     * removes the item from the list
+     * keyfield is used to determine uniqueness of the item added
+     * @param item 
+     * @param keyField 
+     */
     removeItem(item: object, keyField: string) {
         if (this.contatinsItem(item[keyField], keyField)) {
-
+            this.items.splice(this.items.indexOf(item), 1)
         }
     }
 
+
+    /**
+     * checks to see if a particular item exists by the provied key
+     * keyfield is used to determine uniqueness of the item added
+     * @param key 
+     * @param keyField 
+     */
     contatinsItem(key: string, keyField: string): boolean {
         let item = this.items.find(o => o[keyField] === key);
         return item !== null && item !== undefined;
     }
 
+    /**
+     * Clears the list of items
+     */
     clearItems() {
-    
-        this.items = [];
+        while (this.items.length > 0) {
```

Comment:
@thomastighe Why? This seems like it would be more performance intensive. Do you need to save the array reference or something?

Reply:
I do and my thought was that the selected items will probably be fairly low so the performance hit should be minimal.

Reply:
Also in the current place that it is called it will only have a single item at most in it so the memory hit will be very low. I left it as a public helper method in case someone wanted to use it.